### PR TITLE
KAFKA-4764: Upgrade notes for authentication failure handling (KIP-152)

### PR DESCRIPTION
**Pull request by rajinisivaram**

### CODE REVIEW

**Comment by rajinisivaram** [comment]
@ijuma Can you review please? Thank you.

File: docs/upgrade.html [hunk 1]
```diff
@@ -68,6 +68,8 @@
     <li>The Java clients and tools now accept any string as a client-id.</li>
     <li>The deprecated tool <code>kafka-consumer-offset-checker.sh</code> has been removed. Use <code>kafka-consumer-groups.sh</code> to get consumer group details.</li>
     <li>SimpleAclAuthorizer now logs access denials to the authorizer log by default.</li>
+    <li>Authentication failures are now reported to clients as one of the subclasses of <code>AuthenticationException</code>.
+        No retries will be performed if client connection fails authentication.</li>
```
Comment (ijuma) [inline docs/upgrade.html]: Nit: `if a client connection`?

File: docs/upgrade.html [hunk 2]
```diff
@@ -85,6 +87,9 @@
     <li> <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-112%3A+Handle+disk+failure+for+JBOD">KIP-112</a>: MetadataResponse v5 introduces a partition-level <code>offline_replicas</code> field. </li>
     <li> <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-112%3A+Handle+disk+failure+for+JBOD">KIP-112</a>: ProduceResponse v4 introduces error code for KafkaStorageException. </li>
     <li> <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-112%3A+Handle+disk+failure+for+JBOD">KIP-112</a>: FetchResponse v6 introduces error code for KafkaStorageException. </li>
+    <li> <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-152+-+Improve+diagnostics+for+SASL+authentication+failures">KIP-152</a>:
+         SaslAuthenticate request has been added to enable reporting of authentication failures. This request will
+         be used if SaslHandshake request version is greater than v0. </li>
```
Comment (ijuma) [inline docs/upgrade.html]: Nit: `if the SaslHandshake..` and maybe `v0` -> `0` since `version` is mentioned explicitly?

**Review by ijuma — APPROVED** [review]
LGTM. A couple of nits, no re-review needed.

**Comment by rajinisivaram** [comment]
Updated, merging to trunk.

**Pull request closed without merging**